app/controllers/user_targets_controller.rb

```diff
@@ -38,4 +39,31 @@ def load_target
   def update_first_target
     current_user.update(first_target: true) unless current_user.first_target
   end
+
+  def check_new_matches(new_target)
```

Comment:
This whole logic should be in a service, you can take a look at https://www.engineyard.com/blog/keeping-your-rails-controllers-dry-with-services